packages/flutter_tools/lib/src/android/migrations/min_sdk_version_migration.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../project.dart';
+import '../gradle_utils.dart';
+
+@visibleForTesting
+const String flutterMinSdk = 'minSdkVersion flutter.minSdkVersion';
+@visibleForTesting
+const String appGradleNotFoundWarning = 'Module level build.gradle file not found, skipping minSdkVersion migration.';
+
+class MinSdkVersionMigration extends ProjectMigrator {
+  MinSdkVersionMigration(
+      AndroidProject project,
+      super.logger,
+  ) : _project = project;
+
+  final AndroidProject _project;
+
+  @override
+  void migrate() {
+    try {
+      processFileLines(_project.appGradleFile);
```

Comment:
Can you confirm that on a fresh checkout of a project that needs this migration that a build.gradle is normally present when this line of code runs? The reason I ask is for some plugins we need to run flutter build --config-only to create the build.gradle file first.

Reply:
Confirmed that it does exist (and is 16) on an old commit (ab2b0851a26f0b93d01b4eec608820eb6d7fc41b, if you're curious)